universal-application-tool-0.0.1/test/services/PathTest.java

```diff
@@ -72,6 +72,32 @@ public void keyName() {
     assertThat(path.keyName()).isEqualTo("name");
   }
 
+  @Test
+  public void isList() {
+    Path path = Path.create("one.two[3]");
+    assertThat(path.isList()).isTrue();
+  }
+
+  @Test
+  public void withoutArrayIndex() {
+    Path path = Path.create("one.two[3]");
+    assertThat(path.withoutArrayIndex()).isEqualTo(Path.create("one.two"));
```

Comment:
suggestion (non-blocking): I thought "withoutArrayIndex" just removed the index, so you would have one.two[]. If that's not the case, could we change the name to something like `withoutList`?

Reply:
I renamed it to `withoutArrayReference`.